src/openms/include/OpenMS/FORMAT/QcMLFile.h

```diff
@@ -166,8 +177,13 @@ namespace OpenMS
     void existsRunQualityParameter(const String filename, const String qpname, std::vector<String>& ids) const;
     ///Returns the ids of the parameter name given if found in given set, empty else
     void existsSetQualityParameter(const String filename, const String qpname, std::vector<String>& ids) const;
+    ///Calculation and collection of QC data
+    void collectQCData(const String& inputfile_id, const String& inputfile_feature,
+                       const String& inputfile_consensus, const String& inputfile_raw,
+                       const bool remove_duplicate_features, ControlledVocabulary cv, MSExperiment exp);
```

Comment:
Do you need to copy the whole experiment? Looks a bit costly but you might have reasons.

Reply:
The copy was not necessary, I changed it for for the mzQC file to const reference.
For the QcML file there seems to be still some issues with const reference, I will look into that.

Reply:
now the experiment gets passed by reference in both mzQC and qcML files